[jamieeeleow] iP

jamieeeleow opened the pull request:
# **Leo**
Your #1 Tasks Manager
> “Nothing is particularly hard when you divide it into small jobs." - [Henry Ford](https://dansilvestre.com/productivity-quotes/)

Overwhelmed by _the pile of work, the ticking deadlines and the numerous events_ in life? Fred not, Leo is here to manage your tasks! 🗒️ 😃 


### Leo is:
- **free** to download
- **easy** to learn
- **fast** to use
- **clear** and **concise** user interface
- constantly **improving** to serve all your needs
- designed for the ***best*** user experience


### What are you waiting for?
Download Leo now! 👇 
1. Click the [link](https://github.com/nus-cs2103-AY2223S2/ip/pull/103) provided.
2. Download the application.
3. Add your tasks. 


### Features
- [x] Managing your tasks
- [ ] Managing your deadlines <sub>coming soon!
- [ ] Managing your event schedules <sub>coming soon!
- [ ] Reminders <sub>coming soon!


If you’re up for a challenge, code your own version of Leo!!
Here’s the `Leo` class to start you off:
```ruby
public class Leo {

    private final Ui ui;

    public static void main(String[] args) {
        new Leo("data/leo.txt");
    }

    private Leo(String filePath) {
        ui = new Ui();
        run();
        try {
            Storage storage = new Storage(filePath);
            Parser parser = new Parser(storage);
            parser.readCommand();
        } catch (LeoException e) {
            Ui.displayMessage(e.getMessage());
        }
    }

    private void run() {
        ui.greetings();
    }

}
```

Commit:
Add Gradle support

jamieeeleow committed:
Implemented skeletal version of Leo.

jamieeeleow committed:
Implemented the ability to add and display tasks.

jamieeeleow committed:
Changed the implementation of Leo to make use of Storage using some OOP.

jamieeeleow committed:
Implemented ability to mark Task as done.

jamieeeleow committed:
Implemented ability to unmark Task.

jamieeeleow committed:
Implemented a Task class to represent the task and status.

jamieeeleow committed:
Modified to complement new mark and unmark features, and new Task class.

jamieeeleow committed:
Added support to track three types of task: ToDos, Deadlines and Events.

jamieeeleow committed:
Used inheritance to support different types of tasks.

jamieeeleow committed:
Modified class to complement addition of different types of tasks.

jamieeeleow committed:
Taught Leo how to handle errors when dealt with invalid inputs from users.

jamieeeleow committed:
Used LeoException class to represent exceptions specific to Leo.

jamieeeleow committed:
Modified class to throw, handle and display instances of LeoException.

jamieeeleow committed:
Added feature to delete task from list.

jamieeeleow committed:
Added error handling for when task does not exist in list.

jamieeeleow committed:
Included ListAction class for Mark, Unmark and Delete class to share common function.

jamieeeleow committed:
Modify class to accommodate new Delete feature.

jamieeeleow committed:
Used the input/output redirection technique to semi-automate the testing of Leo.

jamieeeleow committed:
Cleaned up import statements according to Java Coding Standards.

jamieeeleow committed:
Implemented a hard disk to save the tasks list.

jamieeeleow committed:
Implemented a hard disk to save the tasks list.

jamieeeleow committed:
Taught Leo how to understand dates and time.

jamieeeleow committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Taught Leo how to understand dates and time.

# Conflicts:
#	src/main/java/Storage/Deadline.java
#	src/main/java/Storage/Event.java

jamieeeleow committed:
Made the code more OOP.

jamieeeleow committed:
Made the code more OOP.

jamieeeleow committed:
Used enumerations.

jamieeeleow committed:
Organised into Packages.

jamieeeleow committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

jamieeeleow committed:
Automated building of project using Gradle.

jamieeeleow committed:
Automated building of project using Gradle.

jamieeeleow committed:
Added JUnit tests to test the behavior of the code.

jamieeeleow committed:
Packaged the app as an executable JAR file.

jamieeeleow committed:
Added JavaDoc comments to the code.

jamieeeleow committed:
Tweaked the code to comply with coding standard.

jamieeeleow committed:
Added find task feature using keyword.

jamieeeleow committed:
Resolved merge conflicts.

jamieeeleow committed:
Resolved merge conflicts.

jamieeeleow committed:
Included JavaDoc for Find feature.

jamieeeleow committed:
Corrected coding style violations that were detected by checkStyle.

jamieeeleow committed:
Corrected coding style violations that were detected by checkStyle.

cydtseng commented on src/main/java/leo/storage/EventTask.java:
```diff
@@ -0,0 +1,77 @@
+     * @param dtTwo Second LocalDateTime to be compared.
+     * @return Both LocalDateTime are on the same day.
+     */
+    private boolean sameDate(LocalDateTime dtOne, LocalDateTime dtTwo) {
```
Great method names, they adhere to the Java coding standard! Perhaps for this method, isSameDate could be something to consider?

cydtseng commented on src/test/java/leo/storage/EventTaskTest.java:
```diff
@@ -0,0 +1,41 @@
+        assertEquals(expectedOne, eventOne.saveFormat());
+        assertEquals(expectedTwo, eventTwo.saveFormat());
+
+    }
```
I like the set of comprehensive test cases you have written!

cydtseng reviewed (commented):
LGTM!:+1:

jamieeeleow committed:
Attempted to add a GUI to Leo.

clydelhui commented:
Very clean code with very good comments and method/attribute names! Code is very easy to follow and understand :)

clydelhui commented on src/main/java/leo/command/ExitCommand.java:
```diff
@@ -0,0 +1,21 @@
+     */
+    public ExitCommand(Storage s, String c) {
+        super(s, c);
+        exit();
```
Seems like this is the only method that calls
`command.exit()`

Perhaps you can consider putting the implementation of `exit()` within the `ExitCommand` class and making it a private method to ensure that other commands are not able to call `exit()`?

clydelhui commented on src/main/java/leo/Leo.java:
```diff
@@ -0,0 +1,36 @@
+
+    private void run() {
+        ui.greetings();
+    }
```
Perhaps a different name for this function could be considered? `initialText()` or something similar?

clydelhui commented on src/main/java/leo/command/Command.java:
```diff
@@ -0,0 +1,50 @@
+
+/**
+ * Represents a command input by user.
+ */
+public class Command {
+
```
Would it be better if this class is made abstract so that it can't be instantiated?

clydelhui commented on src/main/java/leo/storage/Task.java:
```diff
@@ -0,0 +1,105 @@
+public class Task {
+
+    private final String task;
+    private boolean done;
```
Perhaps `isDone` could be considered for this boolean?

clydelhui commented on src/main/java/leo/storage/Task.java:
```diff
@@ -0,0 +1,105 @@
+/**
+ * Represents a Task. A <code>Task</code> object contains the String description.
+ */
+public class Task {
```
Perhaps `task` could be an abstract class as well? Making the other methods abstract will allow you to ensure that the implementation in the child classes will be used.

clydelhui reviewed (commented):
Very neat code with good comments and easy to read and understand variable and method names! Looks like your code follows the coding standard! 👍

chongweiguan commented on src/main/java/leo/storage/Storage.java:
```diff
@@ -0,0 +1,238 @@
+                        String[] temp = description.split("\\|");
+                        String taskDescription = temp[0].strip();
+                        String time = temp[1].strip();
+                        LocalDateTime dt = convertString(time);
```
Very very small detail but maybe make the variable name more clear? dateTime instead of dt?

chongweiguan commented on src/main/java/leo/storage/Storage.java:
```diff
@@ -0,0 +1,238 @@
+                            taskList.add(t);
+                        } else if (taskType == 'E') {
+                            String to = temp[2].strip();
+                            LocalDateTime dtTo = convertString(to);
```
Same here! make the variable name clearer

chongweiguan reviewed (commented):
Hey jamie! Im your commenter and im supposed to review your code quality. Honestly you are doing great, cant seem to find any major issues here. Keep up the good work!

jamieeeleow committed:
Added a GUI for Leo.

jamieeeleow committed:
Updated Test Files based on new files added.

jamieeeleow committed:
Used assert feature to document important assumptions.

jamieeeleow committed:
Improve code quality

Quality of code is bad with different layers of abstraction and deep nesting.

Poor code quality makes it hard to understand and debug.

Cleaning up the code will allow better understandability.

Let's use SLAP and avoid arrowhead style code.

jamieeeleow committed:
Merge pull request #2 from jamieeeleow/branch-A-Assertions

Assertions

jamieeeleow committed:
Merge pull request #1 from jamieeeleow/branch-A-CodeQuality

Improve code quality

jamieeeleow committed:
Add extension to view schedules on a particular day

jamieeeleow committed:
Modify GUI of Leo.

jamieeeleow committed:
Improve code to handle more errors

jamieeeleow committed:
Write more JUnit tests

jamieeeleow committed:
Add Ui.png to docs

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Add images to use for README.md

jamieeeleow committed:
Modify string display of FindCommand.java

jamieeeleow committed:
Merge remote-tracking branch 'origin/master'

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Clean Code Quality based on SOC feedback

jamieeeleow committed:
Revert "Clean Code Quality based on SOC feedback"

This reverts commit a8cde8f8a60b8cf3bbd69b81b43ced387ad08606.

jamieeeleow committed:
Merge commit 'd1894cb3da72a1798e502f495d6c63f9f0dad875'

* commit 'd1894cb3da72a1798e502f495d6c63f9f0dad875':
  Update README.md

jamieeeleow committed:
Clean Code Quality based on SOC feedback

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Update README.md

Attempt to link images

jamieeeleow committed:
Update README.md

Include raw link of images

jamieeeleow committed:
Update README.md

jamieeeleow committed:
Modify checkstyle of test code

jamieeeleow committed:
Create file if file does not exist

jamieeeleow committed:
Revert "Create file if file does not exist"

This reverts commit e2a03cc913ac55f9ab1134df7f8824abe678860f.

jamieeeleow committed:
Modify code so it is not hardcoded

jamieeeleow committed:
Modify code so it is not hardcoded

jamieeeleow committed:
Modify code so it is not hardcoded
